### ISSUE TITLE
[regression] MSVC Warning C4459: declaration of 'indices' hides global declaration

### ISSUE THREAD

**Issue by wanghan02** [boostorg/geometry#1169]
There is a fixed [ticket#835](https://github.com/boostorg/geometry/issues/835). Now the warning pops out again with boost 1.82.0 at a different location. MSVC version is VS2022 17.1.0.

```
boost_1_82_0\boost\geometry\algorithms\detail\overlay\colocate_clusters.hpp(89,1): warning C4459: declaration of 'indices' hides global declaration
boost_1_82_0\boost\multi_array\base.hpp(70,32): note: see declaration of 'boost::`anonymous-namespace'::indices'
boost_1_82_0\boost\geometry\algorithms\detail\overlay\handle_colocations.hpp(100): note: see reference to function template instantiation 'void boost::geometry::detail::overlay::colocate_clusters<Clusters,Turns>(const Clusters &,Turns &)' being compiled
        with
        [
            Clusters=cluster_type,
            Turns=turn_container_type
        ]
```

**Comment by wanghan02** [comment]
This is still a problem in boost 1.83.0.

**Comment by barendgehrels** [comment]
Fixed by #1202
